apps/modernization-ui/src/apps/system-management/components/case-report-and-lab/CaseReportLaboratorySection.spec.tsx

```diff
@@ -0,0 +1,70 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { CaseReportLaboratorySection } from './CaseReportLaboratorySection';
+
+const mockLinks = [
+    { text: 'Manage trigger codes for case reporting', href: '/trigger' },
+    { group: 'Manage lab results' },
+    { text: 'Manage SNOMEDs', href: '/snomeds' }
+];
+
+describe('CaseReportLaboratorySection', () => {
+    const setup = (filter = '') => {
+        const setAlert = jest.fn();
+        render(<CaseReportLaboratorySection filter={filter} setAlert={setAlert} links={mockLinks} />);
+        return { setAlert };
+    };
+
+    it('renders filtered links based on the filter prop', () => {
+        setup('SNOMED');
+        expect(screen.queryByText(/trigger/i)).not.toBeInTheDocument();
+        expect(screen.getByText(/Manage SNOMEDs/i)).toBeInTheDocument();
+    });
+
+    it('does not render component if no matching links', () => {
+        const { container } = render(
+            <CaseReportLaboratorySection filter="xyz" setAlert={jest.fn()} links={mockLinks} />
+        );
+        expect(container).toBeEmptyDOMElement();
+    });
+
+    it('shows confirmation modal when button is clicked', () => {
+        setup();
+        fireEvent.click(screen.getByRole('button', { name: /reset lab mapping cache/i }));
+        expect(screen.getByText(/Are you sure/i)).toBeInTheDocument();
+    });
+
+    it('calls setAlert with success after confirming reset', async () => {
+        const { setAlert } = setup();
+        global.fetch = jest.fn(() =>
+            Promise.resolve({
+                ok: true
+            })
+        ) as jest.Mock;
+
+        fireEvent.click(screen.getByRole('button', { name: /reset lab mapping cache/i }));
+        fireEvent.click(screen.getByRole('button', { name: /yes, reset/i }));
+
```

Comment:
Should use `userEvent` over fireEvent.

https://testing-library.com/docs/user-event/intro/